### PR TITLE
deps: upgrade x/tools and gopls to 8fe064f8

### PR DESCRIPTION
**Pull request by myitcv**
* internal/lsp/protocol: actually handle cancellation delivery 8fe064f8
* internal/lsp: remove the checkErrors command in internal/lsp/source bdfa187a
* internal/lsp: push more build-specific logic into the view e0332898
* internal/lsp: reload metadata for orphaned files e23f2f3a
* internal/lsp: disable literal completion candidates for some clients 1b668f20
* internal/lsp: small change to helper.go to use ast.IsExported 3f4d10fc
* go/analysis/passes/asmdecl: add support for riscv64 628e9aa3
* internal/lsp/cache: move mod-related functions and file e0a7ba33
* internal/lsp: generate boilerplate stubs for type Server 9778d966
* internal/lsp: stop returning errors when we can't find a snapshot 5c352bb4
* internal/lsp/cache: check go.mod even if tempModFile is false a4b4a673
* internal/lsp/cache: fix GOPATH vendoring 5ecc1643
* go/packages: fix GOPATH vendoring with overlays ce9bf563
* internal/lsp: eliminate redundant view.ModFile function 24841a4f
* internal/lsp: recreate the view when needed 59ae353e
* internal/lsp: check that a file handle is unmodified before read e873952e
* internal/lsp: batch file changes in didChangeWatchedFiles 219d3418
* internal/lsp: support batched on-disk changes in source.DidModifyFiles e54d0edf
* internal/lsp: support multiple URIs in (*view).invalidateContent a356fb7f
* internal/lsp: use x/mod to get edits for go.mod quick fixes 0043dadf
* internal/imports: add buildflags to ProcessEnv 2b7b26d7
* go/packages: fix doc for NeedExportsFile 45e69182
* go/packages: handle an overlay edge case with test variants 593de606
* internal/lsp: add tests for references includeDeclaration setting ba161d9e
* internal/lsp: refactor (*snapshot).clone to handle multiple URIs 18389cb1
* go/packages: refactor list driver e0f1ed8c
* cmd/stress: use an ellipsis when truncating output bf1340f1
* internal/lsp/cmd: add a test for client logging fb353ca4
* internal/lsp/source: support dereferencing for completion 9bae6688
* go/packages: remove named query 7b0c362a
* gopls/integration/govim: update to latest govim 0a06c699
* go/ssa: replace DefaultType with go/types.Default 213b5e13
* go/ast/inspector: fix misnomer in callback signature dc16b668
* analysis/passes: consolidate imports helper f88bd705
* internal/lsp: remove the Context argument from NewSession 78d06742
* internal/telemetry/log: correct the docstring for Error 29f64efd
* internal/lsp/protocol: handle cancellation delivery b3205ff6
* internal/lsp: eliminate getFileLocked function fc669b13
* internal/lsp/source: rename "typeInference" to "candidateInference" 3e306b7b

### CODE REVIEW

**Comment by myitcv** [comment]
Will merge on a green build because the only change on the `govim` side here is to re-enable (and slightly fix up) a test that now passes because of changes in `gopls`

File: cmd/govim/testdata/scenario_default/quickfix_new_files.txt [hunk 1]
```diff
@@ -84,7 +84,7 @@ func TestDoIt(t *testing.T) {
 -- errors.golden --
 [
   {
-    "bufnr": 1,
+    "bufname": "main.go",
```
Comment (leitzler) [inline cmd/govim/testdata/scenario_default/quickfix_new_files.txt]: Test still calls `vimexprwait errors.golden getqflist()` instead of the new ones so this will fail when enabled.

Comment (myitcv) [reply cmd/govim/testdata/scenario_default/quickfix_new_files.txt]: Thanks - I just flipped this back to the previously proposed order of the stack. So this will merge after #715

**Comment by myitcv** [comment]
Full disclosure - in a later commit I skipped `cmd/govim/testdata/scenario_completeunimported/complete_new_file.txt` because we're seeing that fail on master. Correspondingly I will be re-opening golang.org/issues/36671 shortly
